src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ActivatorUtilities.cs

```diff
@@ -66,10 +64,16 @@ public static class ActivatorUtilities
             {
                 constructors = GetOrAddConstructors(instanceType);
             }
+
+            int maxArgs = GetMaxArgCount();
+            object?[] values = new object?[maxArgs * 2];
+            Span<object?> parameterValues = new(values, 0, maxArgs);
```

Comment:
Can we stack allocate for a small number of args rather than allocating an array?

Reply:
Sure thanks